### PR TITLE
Further fixes for e2e workflow

### PR DESCRIPTION
**Pull request by turkenh**
### Description of your changes

(Hopefully) last set of fixes to make e2e working in CI.

I have:

- [x] Read and followed Upbound's [contribution process](https://git.io/fj2m9).
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR, as appropriate.

### How has this code been tested

By triggering CI with comments below.

### CODE REVIEW

File: test/setup.sh [hunk 1]
```diff
@@ -5,14 +5,23 @@ UPTEST_GCP_PROJECT=${UPTEST_GCP_PROJECT:-crossplane-playground}
 
 echo "Running setup.sh"
 echo "Waiting until configuration package is healthy/installed..."
-${KUBECTL} wait configuration.pkg platform-ref-gcp --for=condition=Healthy --timeout 5m
-${KUBECTL} wait configuration.pkg platform-ref-gcp --for=condition=Installed --timeout 5m
+${KUBECTL} wait configuration.pkg platform-ref-gcp --for=condition=healthy --timeout 5m
```
Comment (ytsarev) [inline test/setup.sh:8]: why do we need to switch cases here?

Just quickly checked status yaml of one of installed Configurations:
```
status:
  conditions:
  - lastTransitionTime: "2022-10-17T08:37:38Z"
    reason: HealthyPackageRevision
    status: "True"
    type: Healthy
```

File: package/cluster/gke/composition.yaml [hunk 2]
```diff
@@ -130,7 +130,7 @@ spec:
             nodeConfig:
               - diskType: pd-standard
                 diskSizeGb: 100
-                imageType: COS
+                imageType: COS_CONTAINERD
```
Comment (ytsarev) [inline package/cluster/gke/composition.yaml:133]: nice find 👍

**Review by ytsarev — COMMENTED** [review]
left couple of comments

File: test/setup.sh (hunk 1 above)
Comment (turkenh) [reply test/setup.sh:8]: No special reason, noticed both works and wanted to go with lowercase consistently. 
From help

```
    --for='':
	The condition to wait on: [delete|condition=condition-name[=condition-value]|jsonpath='{JSONPath
	expression}'=JSONPath Condition]. The default condition-value is true.  Condition values are compared after
	Unicode simple case folding, which is a more general form of case-insensitivity.
```

Comment (turkenh) [reply test/setup.sh]: Reverted not to lead confusion though.

**Review by ytsarev — APPROVED** [review]

**Pull request merged**